sycl/doc/extensions/experimental/sycl_ext_oneapi_device_global.asciidoc

```diff
@@ -643,7 +643,7 @@ global to be initialized immediately before the copy happens.  (Typically, the
 copy operation causes the _device image_ to be loaded onto the device also.)
 As a result, copying from a device global returns the initial value if the
 _device image_ that contains the variable is not currently loaded onto the
-device.
+device. It is undefined behavior to copy to or from a device global on a queue whose device cannot load a _device image_ that references the device global.
```

Comment:
I think this is a valid concern, but it would be better to list this restriction in the section that describes the copy operations "[Add new copy and memcpy members to the handler class](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_device_global.asciidoc#add-new-copy-and-memcpy-members-to-the-handler-class)".

For example, I think we could expand this existing paragraph:

> If `PropertyListT` contains the `device_image_scope` property and the _dest_ variable exists in more than one _device image_ for this queue’s device, the implementation throws an `exception` with the `errc::invalid` error code.

to:

> If `PropertyListT` contains the `device_image_scope` property and the _dest_ variable exists in more than one _device image_ for this queue’s device, the implementation throws an `exception` with the `errc::invalid` error code.
>
> If `PropertyListT` contains the `device_image_scope` property, kernels in the _device image_ containing the _dest_ variable must be compatible with the device that is associated with the `handler`.  If this is not the case, the implementation throws an `exception` with the `errc::kernel_not_supported` error code.

If possible, I'd like to specify an exception rather than saying it is UB.  We already require the `kernel_not_supported` exception if the application attempts to enqueue an incompatible kernel, so I suspect that behavior might fall out for free.  Thoughts?